### DISCUSSION TITLE
How to use SeedExtensions?

### DISCUSSION THREAD

**Discussion by Kralizek** [AutoFixture/AutoFixture#1196] [Q&A]
Hi,

I'm trying to use the extension methods found in the AutoFixture.SeedExtensions package but I am not sure if my expectations are right.

Given this simple test class
```csharp
public class Person
{
	public string FirstName { get; set; }
	
	public string LastName { get; set; }
	
	public int Age { get; set; }
}
```

I wrote these two tests but are both failing
```csharp
[Fact]
public void Person_should_have_firstName_seeded()
{
	var fixture = new Fixture();
	
	var seed = new Person { FirstName = "John" };
	
	var actual = fixture.Create(seed);
	
	Assert.Equal(seed.FirstName, actual.FirstName);
}

[Fact]
public void Person_should_have_age_seeded()
{
	var fixture = new Fixture();

	var seed = new Person { Age = 18 };

	var actual = fixture.Create(seed);

	Assert.Equal(seed.Age, actual.Age);
}

// Completed 2 tests in 0,009s (2 failed)
```

I would assume I am doing something wrong or I totally misunderstood the intent of the extensions.

**Comment by Ergamon** [comment]
I am really lacking the usefullness of this feature, but the extension method just creates a SeededRequest for the specified type.

By default this is just ignored. You need to add a specimenbuilder to your fixture to make it work.

So the shortest version to fix your test would be that:
```
      var fixture = new Fixture();
      fixture.Customize<Person>(c => c
        .FromSeed(s => new Person() { FirstName = s.FirstName })
        .Without(p => p.FirstName));

      var seed = new Person { FirstName = "John" };
      var person = fixture.Create(seed);
```

**Comment by Kralizek** [comment]
Oh thanks!

I was expecting that the seeded `Create` overload was automatically using any property whose value is not `default` without the need to customize the type.

**Comment by ecampidoglio** [comment]
Indeed, the seed feature is not particularly intuitive.

For reference, I wrote about how it works in a [Stack Overflow answer](https://stackoverflow.com/a/28475970/26396) a few years ago.

**Comment by Kralizek** [comment]
Would a PR that changes the need for a customization for the default behavior be welcome?
